### PR TITLE
1477958: Use inotify for checking changes of consumer certs

### PR DESCRIPTION
**Pull request by jirihnidek**
- Bug fix: https://bugzilla.redhat.com/show_bug.cgi?id=1477958
- I-notify is used for checking of changes in /etc/pki/consumer
  to trigger reloading of `identity`.
- When `pyinotify` module or i-notify at all is not possible to use, then it
  falls back to periodical directory polling. This code is used
  by sub-man gui.
- Small changes in subscription-manager.spec file, because new
  Python module is required at RHEL7 and Fedora.

### CODE REVIEW

**Comment by jirihnidek** [comment]
[Bug 1477958 - DBus object Attach does not reflect a changed state of a subscription identity](https://trello.com/c/e5iZJy2h/566-bug-1477958-dbus-object-attach-does-not-reflect-a-changed-state-of-a-subscription-identity)

**Comment by jirihnidek** [comment]
It is necessary to implement disabling of using i-notify in rhsm.conf file, because it will not work at NFS share. It will be also necessary to update documentation (manual pages).

**Comment by jirihnidek** [comment]
Bot, retest this please

File: etc-conf/rhsm.conf [hunk 1]
```diff
@@ -67,6 +67,14 @@ pluginConfDir = /etc/rhsm/pluginconf.d
 # Manage automatic enabling of yum plugins (product-id, subscription-manager)
 auto_enable_yum_plugins = 1
 
+# Inotify is used for monitoring changes in directories with certificates.
+# Currently only /etc/pki/consumer directory is monitored by rhsm.service.
+# When this directory is mounted using network file system without support
+# of inotify notification (e.g. NFS), then it is strongly recommended to
+# disable inotify. When inotify is disabled, then periodical directory
+# polling is used.
```
Comment (awood) [inline etc-conf/rhsm.conf]: The English here is a little off.  I would rephrase as follows:

```
# Inotify is used for monitoring changes in directories with certificates.
# Currently only the /etc/pki/consumer directory is monitored by the rhsm.service.
# When this directory is mounted using a network file system without inotify notification support
# (e.g. NFS), then disabling inotify is strongly recommended. When inotify is disabled, periodical directory
# polling is used instead.
```

File: man/rhsm.conf.5 [hunk 2]
```diff
@@ -181,6 +181,11 @@ pluginConfDir
 .RS 4
 The directory to search for plug-in configuration files
 .RE
+.PP
+inotify
+.RS 4
+Inotify is used for monitoring changes in directories with certificates. Currently only /etc/pki/consumer directory is monitored by rhsm.service. When this directory is mounted using network file system without support of inotify notification (e.g. NFS), then it is strongly recommended to disable inotify. When inotify is disabled, then periodical directory polling is used.
```
Comment (awood) [inline man/rhsm.conf.5]: If this is the same text as the comment in the conf file, then the previous comment should cover this also.

**Comment by awood** [comment]
The unit test failure you're seeing here should be fixed in master (via #1707), so you might want to rebase.

**Comment by kahowell** [comment]
Looks like some of the test failures might be due to:

    'Identity' object has no attribute 'getConsumerId'

That's the only thing that stands out in the test logs to me (that and a couple of assertions, but those are more obvious)...

**Comment by jirihnidek** [comment]
@kahowell thanks, it was very good point. I forgot to replace some methods `getConsumerId()` with new property `uuid`. Class `ConsumerIdentity` still has a method with the same name and it was easy to overlook. My bad. All tests are passing now. I will squash all commits and I will make it ready for merge/review.
